### DISCUSSION TITLE
oauth2 example

### DISCUSSION THREAD

**Discussion by nrw303** [SAP/luigi#3980] [Q&A] [answered]
hi, 
I'm new here and I'm evaluating Luigi as a portal for our apps (we're an SAP build partner). I want to use oauth2 with Microsoft Entra ID as a provider to authenticate my users... and I'm working in Svelte.

Does anyone have an example of how to do something like this? the docs do provide some info about auth but no examples.

thanks in advance!

**Comment by hardl** [comment, marked as answer]
Hi @nrw303,
we have an example config in luigi fiddle, using an IDP mock 
![image](https://github.com/user-attachments/assets/c84051fd-39eb-4be7-8e3b-1e702d69e304)

In general using luigi's auth plugins is not required but rather a convenience option. You can always set up a "traditional" auth mechanism like with any other web app and connect luigis UI parts (mainly the profile menu with username, logout, etc.) using the corresponding config options, e.g. https://docs.luigi-project.io/docs/navigation-parameters-reference?section=profile
